# BASIC PROBLEM STATEMENT
## Masalah yang akan diselesaikan yaitu apakah mayoritas pembeli menggunakan voucher diskon ?, asumsikan bahwa data yang dimiliki yaitu food_delivery_datasets.csv. Maka algoritma nya yaitu :
### 1. Import Dataset dan Library
### 2. Cleaning data dengan status completed
### 3. Gunakan kolom voucher_amount dan status

### 1. Import library
#### library yang digunakan hanya pandas karena saya hanya membutuhkan 'pd.read_csv' untuk membaca data berupa file csv, kemudian mengimpor data yaitu food_delivery_datasets

In [13]:
import pandas as pd
df=pd.read_csv('https://raw.githubusercontent.com/alamhanz/pandas_journey/main/mydata/food_delivery_datasets.csv')

#### kelompokkan data berdasarkan status menggunakan perintah 'groupby', karena data yang akan dikelompokkan adalah food_delivery_datasets.csv yang telah dimisakan dengan 'df', maka perintahnya yaitu "df.groupby('status')"
#### kemudian cetak hasil pengelompokkan tadi dengan perintah "group_status['status'].size()

In [12]:
group_status = df.groupby('status')
group_status['status'].size()

status
Completed    881
Name: status, dtype: int64

#### Cleaning dataset dengan mensortir orderan dengan status 'completed', kemudian index data baru yang hanya memiliki status order 'completed'
#### Karena data sudah dipisahkan dengan status 'Cancelled', maka yang perlu dilakukan adalah mengelompokkan order dengan voucher_amount menggunakan perintah 'groupby', lalu cetak data yang telah diolah

In [14]:
df = df[df.status == 'Completed']
df.reset_index(drop=True)
group_voucher = df.groupby('voucher_amount')
group_voucher['voucher_amount'].size()

voucher_amount
0        743
2000       5
3000      13
4000      13
5000      69
6000       3
7000       1
8000       1
9000       2
10000      9
12000      1
15000     15
16500      1
20000      3
25000      2
Name: voucher_amount, dtype: int64

#### Dari hasil di atas, dapat dilihat bahwa sebanyak 743 dari 881 order tidak menggunakan voucher diskon karena nilai voucher_amount nya 0, sehingga dapat disimpulkan bahwa mayoritas pembeli yang menyelesaikan orderannya (berstatus 'completed') tidak menggunakan voucher diskon